lms/djangoapps/courseware/tests/test_utils.py

```diff
@@ -14,13 +16,44 @@
 from common.djangoapps.student.tests.factories import GlobalStaffFactory, UserFactory
 from lms.djangoapps.courseware.constants import UNEXPECTED_ERROR_IS_ELIGIBLE
 from lms.djangoapps.courseware.tests.factories import FinancialAssistanceConfigurationFactory
+from lms.djangoapps.courseware.access_utils import adjust_start_date
 from lms.djangoapps.courseware.utils import (
     create_financial_assistance_application,
     get_financial_assistance_application_status,
-    is_eligible_for_financial_aid
+    is_eligible_for_financial_aid,
 )
 
 
+@ddt.ddt
+class TestAccessUtils(TestCase):
+    """Tests for the access_utils functions."""
+
+    @ddt.data(
+        # days_early_for_beta, is_beta_user, expected_date
+        (None, True, "2025-01-03"),
+        (2, True, "2025-01-01"),
+        (timedelta.max.days + 10, True, "2025-01-03"),
+        (None, False, "2025-01-03"),
+        (2, False, "2025-01-03"),
+        (timedelta.max.days + 10, False, "2025-01-03"),
+    )
+    @ddt.unpack
+    def test_adjust_start_date(self, days_early_for_beta, is_beta_user, expected_date):
+        """Tests adjust_start_date
+
+        Should only modify the date if the user is beta for the course,
+        and `days_early_for_beta` is sensible number."""
+        start = date(2025, 1, 3)
+        expected = date.fromisoformat(expected_date)
+        user = "princessofpower"
+        course_key = "edx1+8675"
+        with patch("lms.djangoapps.courseware.access_utils.CourseBetaTesterRole") as role_mock:
+            instance = role_mock.return_value
+            instance.has_user.return_value = is_beta_user
+            adjusted_date = adjust_start_date(user, days_early_for_beta, start, course_key)
+            self.assertEqual(expected, adjusted_date)
+
+
```

Comment:
same as above – from here on down it is just autoformatting. Again, I can undo it and just put in the individual changes the reviewers think it is too excessive.